### PR TITLE
[dhcp_server] Append subnet_id to server config and read the id from lease file for STATE_DB updating

### PR DESCRIPTION
**Pull request by w1nda**
#### Why I did it
When parsing lease, we use mac as index to find vlan name in fdb, however, it won't work when there is no such records in fdb.

##### Work item tracking
- Microsoft ADO **(number only)**: 27737469

#### How I did it
Write vlan id to server config as subnet_id, so the subnet_id will be populated to kea-lease.csv, when we parse the kea-lease.csv, we can populate the vlan name.

#### How to verify it
Unit test, pipeline build.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: rules/config [hunk 1]
```diff
@@ -150,7 +150,7 @@ INCLUDE_NAT = y
 INCLUDE_DHCP_RELAY = y
 
 # INCLUDE_DHCP_SERVER - build and install dhcp-server package
-INCLUDE_DHCP_SERVER ?= n
+INCLUDE_DHCP_SERVER ?= y
```
Comment (w1nda) [inline rules/config:153]: Revert this change before merge.

**Review by yaqiangz — APPROVED** [review]

**Comment by w1nda** [comment]
Hi @yxieca, cloud you merge this PR?

**Comment by w1nda** [comment]
Hi @yxieca, cloud you help approve the request for 202311 branch?
thanks in advance.
